plugins/flashrom/fu-plugin-flashrom.c

```diff
@@ -143,15 +151,21 @@ fu_plugin_flashrom_device_set_hwids(FuPlugin *plugin, FuDevice *device)
 }
 
 static gboolean
-fu_plugin_flashrom_coldplug(FuPlugin *plugin, GError **error)
+fu_plugin_flashrom_add_device(FuPlugin *plugin, FuIfdRegion region, GError **error)
 {
+	g_autofree gchar *instance_id = NULL;
+
 	FuContext *ctx = fu_plugin_get_context(plugin);
 	const gchar *dmi_vendor;
-	g_autoptr(FuDevice) device = fu_flashrom_device_new(ctx);
+
+	g_autoptr(FuDevice) device = fu_flashrom_device_new(ctx, region);
 
 	fu_device_set_name(device, fu_context_get_hwid_value(ctx, FU_HWIDS_KEY_PRODUCT_NAME));
 	fu_device_set_vendor(device, fu_context_get_hwid_value(ctx, FU_HWIDS_KEY_MANUFACTURER));
 
+	instance_id = g_strdup_printf("IFD:%s", fu_ifd_region_to_string(region));
+	fu_device_add_instance_id(device, instance_id);
```

Comment:
This isn't really an instance ID. What's here's already (i.e. what's the output of `sudo fwupdtool get-devices --plugins flashrom`)? Could you do something like `fu_device_add_instance_str(device, "REGION", fu_ifd_region_to_string(region));` and then `fu_device_build_instance_id(device, NULL, "FLASHROM", "VENDOR", "PRODUCT", "REGION", NULL);` -- We might also have to define the VENDOR and PRODUCT with `fu_device_add_instance_str` the same way -- and then we need it defined in the plugins/flashrom/README.md file....

Reply:
> This isn't really an instance ID.

Right, sorry, I'm confused about numerous IDs. Read relevant section of the tutorial and sources to improve my understanding.

> This isn't really an instance ID. What's here's already (i.e. what's the output of `sudo fwupdtool get-devices --plugins flashrom`)?

If I update `flashrom.quirk`, I get:
```
$ sudo fwupdtool get-devices --plugins flashrom
NS50MU
│
└─NS50MU:
      Device ID:          84bf22110f952e8493cc6ed99039b756d4129b5c
      Current version:    1.0
      Vendor:             Notebook (DMI:3mdeb Embedded Systems Consulting)
      GUIDs:              230c8b18-8d9b-53ec-838b-6cfc0383493a _ main-system-firmware
                          dc0de287-201b-5240-b8d2-0e5cead707ea _ Notebook&Not Applicable&NS50MU&Not Applicable&Notebook&NS50MU
                          8da9fe2b-90f4-5028-9bd9-c692b107d4ea _ Notebook&Not Applicable&NS50MU&Not Applicable
                          0039fec6-9ecf-5921-9961-905e148e052a _ Notebook&Not Applicable&NS50MU
                          6b26127f-384f-5ecc-9e94-02359da05f54 _ Notebook&Not Applicable&Notebook&NS50MU
                          58b02b86-65a5-56b6-bf24-fd9767043c58 _ Notebook&Not Applicable&NS50MU&Not Applicable&3mdeb Embedded Systems Consulting
      Device Flags:       _ Internal device
                          _ Updatable
                          _ System requires external power source
                          _ Needs shutdown after installation
```

This isn't stock BIOS, but coreboot ported for these laptops:

```
$ fwupdtool hwids
Computer Information
--------------------
BiosVendor: 3mdeb Embedded Systems Consulting
BiosVersion: Dasharo (coreboot+UEFI) v1.0.0
Manufacturer: Notebook
Family: Not Applicable
ProductName: NS50MU
ProductSku: Not Applicable
EnclosureKind: a
BaseboardManufacturer: Notebook
BaseboardProduct: NS50MU

Hardware IDs
------------
not available as 'BiosMajorRelease' unknown
not available as 'BiosMajorRelease' unknown
not available as 'BiosMajorRelease' unknown
{60f53465-e8fc-5122-b79b-f7b03f063037}   <- Manufacturer + Family + ProductName + ProductSku + BaseboardManufacturer + BaseboardProduct
{99a74816-5cd4-5166-b844-df3ef29e7300}   <- Manufacturer + Family + ProductName + ProductSku
{00bb8e01-c72a-5906-aa0e-2279c54ba369}   <- Manufacturer + Family + ProductName
{b05c067a-a96f-591b-9c69-7dda03324e00}   <- Manufacturer + ProductSku + BaseboardManufacturer + BaseboardProduct
{cd9d0da1-e88d-5d36-aae5-2ba60e8d0ed6}   <- Manufacturer + ProductSku
{f42b9241-6cdc-55cb-bb62-393a52752fc8}   <- Manufacturer + ProductName + BaseboardManufacturer + BaseboardProduct
{b7b416f9-55aa-5fd4-bff7-fbaa4143bd92}   <- Manufacturer + ProductName
{b05c067a-a96f-591b-9c69-7dda03324e00}   <- Manufacturer + Family + BaseboardManufacturer + BaseboardProduct
{cd9d0da1-e88d-5d36-aae5-2ba60e8d0ed6}   <- Manufacturer + Family
{a127c2d7-5537-5b72-8bc1-9ccd683f18e6}   <- Manufacturer + EnclosureKind
{774c2cef-39d8-5b40-9477-c220b4fc1f04}   <- Manufacturer + BaseboardManufacturer + BaseboardProduct
{56bb94fc-caf1-5863-9e85-1e449a0c5a01}   <- Manufacturer
```

> and then we need it defined in the plugins/flashrom/README.md file....

You mean describe structure of instance ID there?